planner/core/stats.go

```diff
@@ -40,20 +39,35 @@ func (p *LogicalTableDual) deriveStats() (*property.StatsInfo, error) {
 	return p.stats, nil
 }
 
-func (p *baseLogicalPlan) deriveStats() (*property.StatsInfo, error) {
+func (p *baseLogicalPlan) recursiveDeriveStats() (*property.StatsInfo, error) {
 	if p.stats != nil {
 		return p.stats, nil
 	}
 	if len(p.children) > 1 {
-		panic("LogicalPlans with more than one child should implement their own deriveStats().")
+		err := ErrInternal.GenWithStack("LogicalPlans with more than one child should implement their own recursiveDeriveStats().")
+		return nil, errors.Trace(err)
```

Comment:
`GenWithStack()` generates an error with the call stack. We can just return that error, no need to call `errors.Trace()`.